internal/daemon/controller/handlers/workers/worker_service.go

```diff
@@ -408,6 +416,58 @@ func (s Service) RemoveWorkerTags(ctx context.Context, req *pbs.RemoveWorkerTags
 	return &pbs.RemoveWorkerTagsResponse{Item: item}, nil
 }
 
+// ReadCertificateAuthority will list the next and current certificates for the worker certificate authority
+func (s Service) ReadCertificateAuthority(ctx context.Context, req *pbs.ReadCertificateAuthorityRequest) (*pbs.ReadCertificateAuthorityResponse, error) {
+	const op = "workers.(Service).ReadCertificateAuthority"
+	if err := validateReadCaRequest(req); err != nil {
+		return nil, err
+	}
+
+	authResults := s.authResult(ctx, req.GetScopeId(), action.ReadCertificateAuthority)
+	if authResults.Error != nil {
+		return nil, authResults.Error
+	}
+
+	caProto, err := s.listCertificateAuthorityFromRepo(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	ca := certificateAuthorityToProto(caProto)
+	return &pbs.ReadCertificateAuthorityResponse{Item: ca}, nil
+}
+
+// ReinitializeCertificateAuthority will delete and regenerate the next and current certificates for the worker certificate authority
+func (s Service) ReinitializeCertificateAuthority(ctx context.Context, req *pbs.ReinitializeCertificateAuthorityRequest) (*pbs.ReinitializeCertificateAuthorityResponse, error) {
+	const op = "workers.(Service).ReinitializeCertificateAuthority"
+	if err := validateReinitCaRequest(req); err != nil {
+		return nil, err
+	}
+
+	authResults := s.authResult(ctx, req.GetScopeId(), action.ReinitializeCertificateAuthority)
+	if authResults.Error != nil {
+		return nil, authResults.Error
+	}
+
+	repo, err := s.workerAuthFn()
+	if err != nil {
+		return nil, err
+	}
+
+	err = server.ReinitializeRoots(ctx, repo)
```

Comment:
See comment below, but I think this function could return the new roots (by having the function it chains to return them) so you don't need an additional lookup just after this.